subcommand/acl-controller/command.go

```diff
@@ -466,6 +488,66 @@ func (c *Command) upsertBindingRule(consulClient *api.Client, bindingRule *api.A
 	return nil
 }
 
+// upsertAnonymousTokenPolicy ensures that the anonymous ACL token has the correct permissions
+// to allow for WAN federation via mesh gateways.
+// If mesh gateway WAN federation is enabled and the ACL controller is in the primary
+// datacenter then we need to update the anonymous token with service:read and node:read.
+// Tokens are stripped from cross DC API calls so cross DC API calls use the anonymous
+// token. Mesh gateway proxies use the anonymous token to talk cross-DC and they require
+// service:read and node:read.
+// The anonymous token is global so it is replicated from the primary DC to all secondary
+// DCs, which is why we only update it if this is the primary datacenter.
+func (c *Command) upsertAnonymousTokenPolicy(consulClient *api.Client, agentConfig AgentConfig) error {
+	consulDC, primaryDC, err := c.consulDatacenterList(agentConfig)
+	if err != nil {
+		return fmt.Errorf("failed to list Consul datacenters: %w", err)
+	}
+
+	if !agentConfig.DebugConfig.MeshGatewayWANFederationEnabled || consulDC != primaryDC {
```

Comment:
Yes, it would also be helpful to have a field that indicates whether or not the agent is running an enterprise version. There was this thread that mentioned automation tooling and Nomad are parsing this info out of the version string.. but that seems a bit dubious and broke in 1.12.0. https://hashicorp.slack.com/archives/C045NJD17/p1650927945635299

I guess the best place to bring this up would be in #team-consul, thoughts?

Reply:
Yes, it would also be helpful to have a field that indicates whether or not the agent is running an enterprise version. There was this thread that mentioned automation tooling and Nomad are parsing this info out of the version string.. but that seems a bit dubious and broke in 1.12.0. https://hashicorp.slack.com/archives/C045NJD17/p1650927945635299

I guess the best place to bring this up would be in #team-consul, thoughts?

Reply:
> I guess the best place to bring this up would be in #team-consul, thoughts?

Yep!